### PR TITLE
Info and region alert

### PR DESCRIPTION
**Pull request by VikChawla**
**Description**

Added provider info modal, and integrated region alert with dark mode. 

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js [hunk 1]
```diff
@@ -64,6 +67,10 @@ const GetQuotes = () => {
 		[navigation]
 	);
 
+	const renderInfo = useCallback(() => {
+		setShowInfo(!showInfo);
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:71]: This is not needed, whenever you want to toogle `setState` value you can do `setState(value => !value)`

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js [hunk 2]
```diff
@@ -81,6 +88,16 @@ const GetQuotes = () => {
 	return (
 		<ScreenLayout>
 			<ScreenLayout.Header description="Buy ETH from one of our trusted providers. You’ll be securely taken to their portal without leaving the MetaMask app." />
+			<InfoAlert
+				isVisible={showInfo}
+				subtitle={selectedProviderInfo.subtitle}
+				dismiss={() => setShowInfo(!showInfo)}
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:94]: Set it to `false` explicitly

File: app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js [hunk 3]
```diff
@@ -103,6 +120,7 @@ const GetQuotes = () => {
 										onPress={() => handleOnPress(quote)}
 										onPressBuy={() => handleOnPressBuy(quote)}
 										highlighted={quote.providerId === providerId}
+										showInfo={() => renderInfo()}
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/GetQuotes.js:123]: Set it to `true` explicitly

File: app/components/UI/FiatOnRampAggregator/Views/Region.js [hunk 4]
```diff
@@ -36,16 +37,17 @@ const Region = () => {
...
+	const [selectedUnsupportedLocation, setSelectedUnsupportedLocation] = useState({});
 	const [{ data, isFetching, error }] = useSDKMethod('getCountries');
 
 	useEffect(() => {
 		navigation.setOptions(getFiatOnRampAggNavbar(navigation, { title: 'Region' }, colors));
 	}, [navigation, colors]);
 
 	const handleChangeRememberRegion = () => {
-		setRememberRegion((currentRememberRegion) => !currentRememberRegion);
+		setRememberRegion(!rememberRegion);
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/Region.js:50]: Undo this change.

File: app/components/UI/FiatOnRampAggregator/Views/Region.js [hunk 5]
```diff
@@ -110,7 +106,14 @@ const Region = () => {
 				title={strings('fiat_on_ramp_aggregator.region.your_region')}
 				description={strings('fiat_on_ramp_aggregator.region.subtitle_description')}
 			/>
-
+			<RegionAlert
+				isVisible={showAlert}
+				subtitle={`${selectedUnsupportedLocation.emoji}   ${selectedUnsupportedLocation.name}`}
+				dismiss={() => setShowAlert(!showAlert)}
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/Region.js:112]: Set it to `false` explicitly

File: app/components/UI/FiatOnRampAggregator/Views/Region.js [hunk 6]
```diff
@@ -132,28 +135,29 @@ const Region = () => {
...
 						/>
-						<Text black style={styles.checkboxMargin}>
-							{strings('fiat_on_ramp_aggregator.region.remember_region')}
-						</Text>
-					</TouchableOpacity>
+						<TouchableOpacity activeOpacity={1} onPress={handleChangeRememberRegion}>
+							<Text black style={styles.checkboxMargin}>
+								{strings('fiat_on_ramp_aggregator.region.remember_region')}
+							</Text>
+						</TouchableOpacity>
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/Views/Region.js:159]: Why the TouchableOpacity now does not wrap the Checkbox + Text?

File: app/components/UI/FiatOnRampAggregator/components/RegionAlert.tsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+const createStyles = (colors: any) =>
+	StyleSheet.create({
+		box: {
+			backgroundColor: colors.background.default,
+			paddingHorizontal: 20,
+			paddingBottom: 20,
+		},
+		title: {
+			fontSize: 18,
+		},
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/RegionAlert.tsx:21]: This font size is 2 points bigger than `<Text big>`, can you reach out to design team about the possibility of turning this into a reusable font size (eg: `<Text bigger>`)?

File: app/components/UI/FiatOnRampAggregator/components/RegionModal.tsx [hunk 8]
```diff
@@ -73,7 +74,7 @@ const createStyles = (colors: any) =>
 		separator: {
 			height: 1,
 			width: '100%',
-			backgroundColor: colors.border.muted,
+			backgroundColor: colors.background.alternative,
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/RegionModal.tsx:77]: Do you have a graphic reference about this change?

File: app/components/UI/FiatOnRampAggregator/components/RegionModal.tsx [hunk 9]
```diff
@@ -284,6 +288,14 @@ const RegionModal: React.FC<Props> = ({
 				<ModalDragger />
 				{activeView === RegionViewType.COUNTRY ? (
 					<ScreenLayout>
+						<RegionAlert
+							isVisible={showAlert}
+							subtitle={`${selectedCountryInTransit.emoji}   ${selectedCountryInTransit.name}`}
+							dismiss={() => setShowAlert(!showAlert)}
```
Comment (wachunei) [inline app/components/UI/FiatOnRampAggregator/components/RegionModal.tsx:294]: Set it to `false` explicitly

**Review by wachunei — CHANGES_REQUESTED** [review]
Left a few comments with small changes

File: app/components/UI/FiatOnRampAggregator/Views/Region.js (hunk 6 above)
Comment (VikChawla) [reply app/components/UI/FiatOnRampAggregator/Views/Region.js:159]: It was causing errors before. The touchable opacity and the checkbox were becoming un-synced. Do you want me to try finding another workaround?

File: app/components/UI/FiatOnRampAggregator/components/RegionModal.tsx (hunk 8 above)
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/components/RegionModal.tsx:77]: Yes, change it back. Semantically the separator is should corresponde to a `border.muted` color, we can ask a design review later once we code frees.

File: app/components/UI/FiatOnRampAggregator/Views/Region.js (hunk 6 above)
Comment (wachunei) [reply app/components/UI/FiatOnRampAggregator/Views/Region.js:159]: Make the checkbox a controlled component, remove the `onValueChange` prop and let the wrapper capture the press event.:

example:
```tsx
	<TouchableOpacity onPress={handleChangeRememberRegion} style={styles.checkboxView}>
		<CheckBox
			value={rememberRegion}
			boxType="square"
			animationDuration={0.2}
			onAnimationType="fill"
			offAnimationType="fill"
			onFillColor={colors.primary}
			onCheckColor={colors.background.default}
			style={styles.checkbox}
		/>
		<Text black style={styles.checkboxMargin}>
			{strings('fiat_on_ramp_aggregator.region.remember_region')}
		</Text>
	</TouchableOpacity>
```

Also, if you are going to dismiss the opacity effect with `activeOpacity={1}`, consider using `<Pressable>` instead.

**Review by CryptoGodfatherVA23 — APPROVED** [review]

**Review by wachunei — APPROVED** [review]

**Pull request closed without merging**